src/imgui/ImGuiMachine.cc

```diff
@@ -190,37 +208,7 @@ void ImGuiMachine::showMenu(MSXMotherBoard* motherBoard)
 			saveSetupOpen = im::Menu("Save current setup as ...", true, [&]{
 				ImGui::TextUnformatted("Save current setup:");
 
-				using enum SetupDepth;
-
-				static constexpr array_with_enum_index<SetupDepth, zstring_view> helpText = {
-					zstring_view(""), // NONE
-					"Only the machine itself, without anything in it.",
-					"The machine with all plugged in extensions.",
-					"The machine, with all plugged in extensions and all things that are plugged into the connectors.",
-					"The machine, with all plugged in extensions and all things that are plugged into the connectors and also all inserted media.",
-					"The full state of the machine, with everything that's in it at the current time."
-				};
-
-				auto depthNodeNameForCombo = [&](SetupDepth depth){
-					return tmpStrCat(depth == MACHINE ? "" : "+ ", depthNodeNames[depth]).c_str();
-				};
-
-				im::Combo("Select level", depthNodeNameForCombo(saveSetupDepth), [&]{
-					static const auto indent = ImGui::CalcTextSize("m").x;
-					//for (const auto& d : { MACHINE, EXTENSIONS, CONNECTORS, MEDIA, COMPLETE_STATE }) {
-					for (auto d_ = static_cast<uint8_t>(MACHINE); d_ < static_cast<uint8_t>(NUM); ++d_) {
-						const auto d = static_cast<SetupDepth>(d_);
-						if (d != MACHINE) {
-							ImGui::Indent(indent);
-						}
-						if (ImGui::Selectable(depthNodeNameForCombo(d))) {
-							saveSetupDepth = d;
-						}
-						simpleToolTip(helpText[d]);
-					}
-				});
-				HelpMarker("Select the level to include in the setup that will be saved. "
-					"All levels above the one you selected will also be included.");
+				showSetupLevelSelector("Select level", false, saveSetupDepth);
```

Comment:
Reminder. For sub menus we don't include `...` in the text. Only for items that open a new separate window.
But I now see that for `{Load,Save} {state/replay}` we also do this "wrong", we should also fix those.

Reply:
Fixed this now all over the GUI, using the MS UI guidelines for this as much as I could.